Wangscape/noise/module/ReseedableOps.cpp

```diff
@@ -0,0 +1,502 @@
+    displace_p->module.SetZDisplaceModule(z_displace->getModule());
+
+    return makeModuleGroup({
+        {"source", source},
+        {"displaceX", x_displace},
+        {"displaceY", y_displace},
+        {"displaceZ", z_displace},
+        {DEFAULT_MODULE_GROUP_OUT, displace_p}
+    });
+}
+
+ReseedablePtr translate(ReseedablePtr source, double x_displace, double y_displace, double z_displace)
+{
+    auto translate_point_p = std::make_shared<Reseedable<TranslatePoint>>();
+    translate_point_p->module.SetSourceModule(0, source->getModule());
+    translate_point_p->module.SetXTranslation(x_displace);
+    translate_point_p->module.SetYTranslation(y_displace);
+    translate_point_p->module.SetZTranslation(z_displace);
+
+    return makeModuleGroup({
+        {"source", source},
+        {DEFAULT_MODULE_GROUP_OUT, translate_point_p}
+    });
+}
+
+ReseedablePtr scalePoint(ReseedablePtr source, double x_scale, double y_scale, double z_scale)
+{
+    auto scale_point_p = std::make_shared<Reseedable<ScalePoint>>();
+    scale_point_p->module.SetSourceModule(0, source->getModule());
+    scale_point_p->module.SetXScale(x_scale);
+    scale_point_p->module.SetYScale(y_scale);
+    scale_point_p->module.SetZScale(z_scale);
+
+    return makeModuleGroup({
+        {"source", source},
+        {DEFAULT_MODULE_GROUP_OUT, scale_point_p}
+    });
+}
+
+ReseedablePtr rotatePoint(ReseedablePtr source, double x_angle, double y_angle, double z_angle)
+{
+    auto rotate_point_p = std::make_shared<Reseedable<RotatePoint>>();
+    rotate_point_p->module.SetSourceModule(0, source->getModule());
+    rotate_point_p->module.SetAngles(x_angle, y_angle, z_angle);
+
+    return makeModuleGroup({
+        {"source", source},
+        {DEFAULT_MODULE_GROUP_OUT, rotate_point_p}
+    });
+}
+
+ReseedablePtr turbulence(ReseedablePtr source, double frequency, double power, int roughness, int seed)
+{
+    auto turbulence_p = std::make_shared<Reseedable<Turbulence>>();
+    turbulence_p->module.SetFrequency(frequency);
+    turbulence_p->module.SetPower(power);
+    turbulence_p->module.SetRoughness(roughness);
+    turbulence_p->module.SetSeed(seed);
+    turbulence_p->module.SetSourceModule(0, source->getModule());
+
+    return makeModuleGroup({
+        {"source", source},
+        {DEFAULT_MODULE_GROUP_OUT, turbulence_p}
+    });
+}
+
+ReseedablePtr abs(ReseedablePtr source)
+{
+    auto abs_p = std::make_shared<Reseedable<Abs>>();
+    abs_p->module.SetSourceModule(0, source->getModule());
+
+    return makeModuleGroup({
+        {"source", source},
+        {DEFAULT_MODULE_GROUP_OUT, abs_p}
+    });
+}
+
+ReseedablePtr max(ReseedablePtr a, ReseedablePtr b)
+{
+    auto max_p = std::make_shared<Reseedable<Max>>();
+    max_p->module.SetSourceModule(0, a->getModule());
+    max_p->module.SetSourceModule(1, b->getModule());
+
+    return makeModuleGroup({
+        {"source0", a},
+        {"source1", b},
+        {DEFAULT_MODULE_GROUP_OUT, max_p}
+    });
```

Comment:
There's no reasonable way to compare two modules. This returns a `Max` module returning the pointwise greatest value of the two input modules. This function will be deleted anyway.